Assets/Scripts/Inventory/FirebaseInventoryManager.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Firebase.Database;
+using System.Threading.Tasks;
+
+public class FirebaseInventoryManager : MonoBehaviour
+{
+    [Tooltip("Leave blank to generate ID from hardware")]
+    [SerializeField]
+    internal string _userID;
+
+    [SerializeField]
+    private InventoryManager _inventoryManager;
+
+    private DatabaseReference _database;
+
+    private void Awake()
+    {
+        if (_userID == "")
+            _userID = SystemInfo.deviceUniqueIdentifier;
+        _database = FirebaseDatabase.DefaultInstance.RootReference;
+    }
+
+    public async void LoadData()
+    {
+        if (_database == null) return;
+
+        var task = await _database.Child("users").Child(_userID).GetValueAsync();
+        if (task != null)
+        {
+            string json = task.GetRawJsonValue();
+
+            PlayerData playerData = JsonUtility.FromJson<PlayerData>(json);
```

Comment:
Problem: this crashes if `json` contains `null` or `""`. An easy way of fixing is by doing the following:
`if(string.IsNullOrEmpty(json)) json = "{}";`

This ensures that it won't be null, but instead an object with default values.